### ISSUE TITLE
[TS-26] Explain how to build from source in the Readme

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskshell#25]
_Bruno Vernay on 2016-03-04T20:54:12Z says:_ 

 I would suggest this Readme:
1.  TaskSH

Thank you for taking a look at tasksh!!
* --
1. # Install
1. ## Install from source
 
`TaskSH` uses CMake:

1. Get the source: `git clone https://git.tasktools.org/scm/ex/tasksh.git`
2. Create a build directory: `cd tasksh ; mkdir build ; cd build`
3. Build: `cmake .. && make`.  The program is in `src/tasksh`
4. Install: `sudo make install`

* You could also run the tests: `make test`
* Clean up:  `make clean` or even delete the build directory or  `git clean -dfX`
* Uninstall: `sudo make uninstall` ... and in the meantime, delete the files listed while installing.
1. # License

Tasksh is released under the MIT license. For details check the LICENSE file.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2016-03-04T20:54:12Z
Modified: 2016-03-16T10:45:53Z
```

**Comment by taskwarrior** [comment]
_Bruno Vernay on 2016-03-04T20:58:58Z says:_ 

 Mardown has been interpreted by Jira.

**Comment by taskwarrior** [comment]
_Bruno Vernay on 2016-03-04T20:59:40Z says:_ 

 Might be even easier with the file itself.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2016-03-04T23:03:38Z says:_ 

 Thank you, but there are issues with the attachments:

1. The build information duplicates that which is already in the INSTALL file, but does not agree with it.  Do we need both?
2. The "`libreadline`" dependency is missing.
3. The line "`sudo make uninstall`" generates an error - cmake does not generate an uninstall target.  Did this work when you tested it?
4. Please use "`git format-patch HEAD^`" to generate a patch file from a local commit in a clone.

You're right, Jira messes up careful formatting and it bites me every time, unless you surround the text with:

{ n o f o r m a t }
Formatted text

{ n o f o r m a t }

Remove the spaces, to actually use this.

**Comment by taskwarrior** [comment]
_Bruno Vernay on 2016-03-05T09:45:07Z says:_ 

 I let you decide if you prefer the "All in readme.md", à la GitHub.  
or a more traditional INSTALL, LICENSE, NEWS ... schema.
I will happily contribute to both, just tell me which one, (INSTALL is almost empty.)

I added the `readline` dependency (on my Fedora, the package is called "readline") maybe we should use a per distrib explanation here?? Is it `libreadline` in Ubuntu?

I made the point about uninstall more clear.

The page http://tasktools.org/ contains a instruction `brew install task`   Is tasksh released in brew? or is it just a copy/paste??

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2016-03-13T13:11:36Z says:_ 

 We have already chosen the traditional INSTALL, LICENSE, NEWS approach, so that is preferred, and duplicating the information is not good.

INSTALL is not "almost empty", it contains 127 lines of text. Are you looking at the correct 1.1.0 branch?

The readline dependency is already listed in the INSTALL file.  It should perhaps be "readline" instead of "libreadline", as the library has one name, and the package names are not consistent.  No need to enumerate them all.

The uninstallation instructions in INSTALL certainly needs improvement.

If you run `brew info tasksh` you can see that there is something there. Someone else does brew packaging, not us, so it may need attention.

**Comment by taskwarrior** [comment]
_Bruno Vernay on 2016-03-16T10:40:55Z says:_ 

 The `1.1.0` branch! I looked at the default `Master`, sorry!
`git checkout 1.1.0` and it is way better.

So first: I followed the INSTALL and could run `tasksh`, it worked and the 1144 tests passed green.

Still, there are some details in `INSTALL`, but I guess it would be more efficient to "make" the uninstall first.  
(I liked the CMake "best practice" to create a build directory in order to be able to "Clean-up" the CMake part)

Good to close ...
